File: core/Params/EmailMessageSpecParams.cs
using System;
using core.Entities.EmailandSMS;
using core.Params;

namespace core.Params
{
    public class EmailMessageSpecParams: ParamPages
    {
        public int? Id { get; set; }
        public string Username {get; set;}
        public string Container {get; set;}
        public string SenderEmail { get; set; }
        public string RecipientEmail { get; set; }
        /*public DateTime? MessageSentFrom {get; set;}
        public DateTime? MessageSentUpto {get; set;}
        public DateTime? MessageRecdFrom {get; set;} 
        public DateTime? MessageRecdUpto {get; set;}
        public string Subject { get; set; }
        */
        public EnumMessageType? MessageTypeId {get; set;}
        //public string ContentsLike {get; set;}

    }
}

File: client2/src/app/shared/params/admin/emailMessageSpecParams.ts
export class EmailMessageSpecParams {
     id?: number;
     username: string='';
     container: string='';
     senderEmail: string='';
     recipientEmail: string='';
     messageTypeId: number=0;
     sort = "messageSent";
     pageIndex = 1;
     pageSize = 1;
     search: string='';

}